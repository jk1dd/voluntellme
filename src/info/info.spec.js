import React from 'react'
import Info from './info'
import { shallow } from 'enzyme'

describe('<Info />', () => {
  it('should not crash', () => {
    const info = shallow(<Info />)
    expect(info).toBeTruthy()
  })

  it('renders things at all', () => {
    const info = shallow(<Info />)
    expect(info.contains(<h1>Information</h1>)).toEqual(true)
  })

  it('renders the proper things', () => {
    const info = shallow(<Info />)
    expect(info).toMatchSnapshot()
  })

  it('has a div with class volunteer-info', () => {
    const info = shallow(<Info />)
    expect(info.find('div').length).toEqual(8)
  })

  it('has the right content', () => {
    const info = shallow(<Info />)
    expect(info.equals(<div className='volunteer-info'>
      <div className='tab-header' id='info-block'>
        <h1>Information</h1>
        <p>Find some thoughts and resources here</p>
      </div>
      <div className='off-the-wall'>
        <div className='who-info'>
          <h2>Who volunteers?</h2>
          <p>People all over the country volunteer. You should too. It doesn't matter what age or stage of life you are in - there is always something to help with!</p>
        </div>
        <div className='what-info'>
          <h2>What sorts of things can you do?</h2>
          <ul>
            <li>Pets</li>
            <li>Homeless</li>
            <li>Elderly</li>
            <li>Social Issues</li>
            <li>Education</li>
            <li>Health</li>
          </ul>
        </div>
        <div className='why-info'>
          <h2>What are some reasons to volunteer?</h2>
          <p>It is good for everyone! It can make you happier and heathier, more connected to others and to your community, and bring you enjoyment. At the same time, it makes your community and country a better place to live in, and those who volunteer are more personally invested in the community and its success.</p>
        </div>
        <div className='when-info'>
          <h2>When should you do it?</h2>
          <p>Anytime is a good time! There are so many different types of activities and ways to get involved, that there is always a way to fit in volunteering in your life. The options range from months-long (or even years-long!) commitments, all the way down to just a few hours on a Saturday afternoon.</p>
        </div>
        <div className='how-info'>
          <h2>Want more info?</h2>
          <a href="https://www.nationalservice.gov/resources/recruitment/volunteering-america-resources">Check out some of the resources and information listed at this site!</a>
        </div>
      </div>
    </div>)).toBeTruthy
  })
})
