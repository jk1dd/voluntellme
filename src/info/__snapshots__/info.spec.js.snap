// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Info /> renders the proper things 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="volunteer-info"
>
    <h1>
        Wondering how to get started?
    </h1>
    <div
        className="who-info"
    >
        <h2>
            Who volunteers?
        </h2>
        <p>
            People all over the country volutneer. You should too.
        </p>
    </div>
    <div
        className="what-info"
    >
        <h2>
            What sorts of things can you do?
        </h2>
        <ul>
            <li>
                Pets
            </li>
            <li>
                Homeless
            </li>
            <li>
                Elderly
            </li>
            <li>
                Social Issues
            </li>
        </ul>
    </div>
    <div
        className="why-info"
    >
        <h2>
            What are some reasons to volunteer?
        </h2>
        <p>
            It is good for you and the country
        </p>
    </div>
    <div
        className="when-info"
    >
        <h2>
            When should you do it?
        </h2>
        <p>
            Anytime is a good time
        </p>
    </div>
    <div
        className="how-info"
    >
        <h2>
            Want more info?
        </h2>
        <a
            href="https://www.nationalservice.gov/resources/recruitment/volunteering-america-resources"
        >
            Lots of resources here
        </a>
    </div>
</div>,
  "nodes": Array [
    <div
      className="volunteer-info"
>
      <h1>
            Wondering how to get started?
      </h1>
      <div
            className="who-info"
      >
            <h2>
                  Who volunteers?
            </h2>
            <p>
                  People all over the country volutneer. You should too.
            </p>
      </div>
      <div
            className="what-info"
      >
            <h2>
                  What sorts of things can you do?
            </h2>
            <ul>
                  <li>
                        Pets
                  </li>
                  <li>
                        Homeless
                  </li>
                  <li>
                        Elderly
                  </li>
                  <li>
                        Social Issues
                  </li>
            </ul>
      </div>
      <div
            className="why-info"
      >
            <h2>
                  What are some reasons to volunteer?
            </h2>
            <p>
                  It is good for you and the country
            </p>
      </div>
      <div
            className="when-info"
      >
            <h2>
                  When should you do it?
            </h2>
            <p>
                  Anytime is a good time
            </p>
      </div>
      <div
            className="how-info"
      >
            <h2>
                  Want more info?
            </h2>
            <a
                  href="https://www.nationalservice.gov/resources/recruitment/volunteering-america-resources"
            >
                  Lots of resources here
            </a>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Info />,
      "_debugID": 5,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 3,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="volunteer-info"
>
          <h1>
                    Wondering how to get started?
          </h1>
          <div
                    className="who-info"
          >
                    <h2>
                              Who volunteers?
                    </h2>
                    <p>
                              People all over the country volutneer. You should too.
                    </p>
          </div>
          <div
                    className="what-info"
          >
                    <h2>
                              What sorts of things can you do?
                    </h2>
                    <ul>
                              <li>
                                        Pets
                              </li>
                              <li>
                                        Homeless
                              </li>
                              <li>
                                        Elderly
                              </li>
                              <li>
                                        Social Issues
                              </li>
                    </ul>
          </div>
          <div
                    className="why-info"
          >
                    <h2>
                              What are some reasons to volunteer?
                    </h2>
                    <p>
                              It is good for you and the country
                    </p>
          </div>
          <div
                    className="when-info"
          >
                    <h2>
                              When should you do it?
                    </h2>
                    <p>
                              Anytime is a good time
                    </p>
          </div>
          <div
                    className="how-info"
          >
                    <h2>
                              Want more info?
                    </h2>
                    <a
                              href="https://www.nationalservice.gov/resources/recruitment/volunteering-america-resources"
                    >
                              Lots of resources here
                    </a>
          </div>
</div>,
        "_debugID": 6,
        "_renderedOutput": <div
          className="volunteer-info"
>
          <h1>
                    Wondering how to get started?
          </h1>
          <div
                    className="who-info"
          >
                    <h2>
                              Who volunteers?
                    </h2>
                    <p>
                              People all over the country volutneer. You should too.
                    </p>
          </div>
          <div
                    className="what-info"
          >
                    <h2>
                              What sorts of things can you do?
                    </h2>
                    <ul>
                              <li>
                                        Pets
                              </li>
                              <li>
                                        Homeless
                              </li>
                              <li>
                                        Elderly
                              </li>
                              <li>
                                        Social Issues
                              </li>
                    </ul>
          </div>
          <div
                    className="why-info"
          >
                    <h2>
                              What are some reasons to volunteer?
                    </h2>
                    <p>
                              It is good for you and the country
                    </p>
          </div>
          <div
                    className="when-info"
          >
                    <h2>
                              When should you do it?
                    </h2>
                    <p>
                              Anytime is a good time
                    </p>
          </div>
          <div
                    className="how-info"
          >
                    <h2>
                              Want more info?
                    </h2>
                    <a
                              href="https://www.nationalservice.gov/resources/recruitment/volunteering-america-resources"
                    >
                              Lots of resources here
                    </a>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Info />,
}
`;
